File: src/main/java/com/ject/studytrip/trip/domain/repository/TripReportRepository.java
package com.ject.studytrip.trip.domain.repository;

import com.ject.studytrip.trip.domain.model.TripReport;
import java.util.List;
import java.util.Optional;

public interface TripReportRepository {
    Optional<TripReport> findById(Long tripReportId);

    List<TripReport> findAllByMemberIdAndDeletedAtIsNullOrderByCreatedAtDesc(Long memberId);

    TripReport save(TripReport tripReport);
}

File: src/main/kotlin/com/ject/studytrip/trip/domain/repository/TripReportRepository.kt
package com.ject.studytrip.trip.domain.repository

import com.ject.studytrip.trip.domain.model.TripReport
import java.util.Optional

interface TripReportRepository {
    fun findById(tripReportId: Long): Optional<TripReport>

    fun save(tripReport: TripReport): TripReport
}
